src/Bicep.Core/Semantics/LocalScopeSymbol.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Syntax;
+
+namespace Bicep.Core.Semantics
+{
+    /// <summary>
+    /// Represents a language scope that declares local symbols. (For example the item or index variables in loops are local symbols.)
+    /// </summary>
+    public class LocalScopeSymbol : Symbol, ILanguageScope
```

Comment:
Why is the scope itself a symbol? Shouldn't a symbol represent an identifier?

Reply:
Yeah, I wasn't entirely happy with this. However, in our current design only classes derived from `DeclaredSymbol` represent an identifier. All others do not. Types are symbols but actually lack a proper identifier that uniquely identifies them - the name is more of a alias/suggestion rather than a proper identifier. `FileSymbol` is kind of like the `LocalScopeSymbol` as well.